subsys/esb/esb.c

```diff
@@ -1527,6 +1539,16 @@ static void radio_irq_handler(void)
 		}
 		nrf_radio_event_clear(NRF_RADIO, ESB_RADIO_EVENT_END);
 	}
+
+#if defined(CONFIG_ESB_FAST_CHANNEL_SWITCHING)
```

Comment:
IS_ENABLED ?

Reply:
RXREADY event not available to all SoCs.
Also, when CONFIG_ESB_FAST_CHANNEL_SWITCHING=n the interrupt handler is faster.